reverse_engineering/helpers/restApiHelper.js

```diff
@@ -118,79 +102,48 @@ const createRestApiService = ({ connectionInfo, app }) => {
-	logger.info(`${bucketName}: Start getting documents using REST API`);
-
-	const apiService = createRestApiService({ connectionInfo, app });
-	const count = await apiService.count(bucketName);
-	const body = await apiService.getRandomKey(bucketName);
-	const size = count ?? DEFAULT_LIMIT;
-
-	if (body.error === 'fallback_to_all_docs') {
-		logger.error({ message: `"localRandomKey" in not available or bucket ${bucketName} is empty` });
+/**
+ * @param {{ connectionInfo: ConnectionInfo; bucketName: string; scopeName: string; collectionName: string; logger: Logger; app: App }} param0
+ * @returns {Promise<Document[]>}
+ */
+const getCollectionDocuments = async ({ connectionInfo, bucketName, scopeName, collectionName, logger, app }) => {
+	try {
+		logger.info(`${bucketName}.${scopeName}.${collectionName}: Start getting documents using REST API`);
+
+		const apiService = createRestApiService({ connectionInfo, app });
+		const { rows } = await apiService.getCollectionDocuments({
+			bucketName,
+			scopeName,
+			collectionName,
+			limit: DEFAULT_LIMIT,
+		});
 
-		return await apiService.getDocuments(bucketName, size);
+		return rows
+			.filter(row => !isBinaryFile(row))
+			.map(row => ({
+				[bucketName]: safeParse(row.doc?.json),
+				docid: row.id,
+			}));
+	} catch (error) {
+		logger.error(error);
+		return [];
 	}
-
-	logger.info(`REST API: fetching documents by random key`);
-
-	const maxNumberOfAttempts = 5;
-
-	let numberOfAttempts = 0;
-
-	return new Promise((resolve, reject) => {
-		async.timesLimit(
-			size,
-			50,
-			async index => {
-				if (numberOfAttempts > maxNumberOfAttempts) {
-					return;
-				}
-
-				const keyBody = await apiService.getRandomKey(bucketName).catch(error => {
-					logger.error({ message: `Error fetching random key from bucket "${bucketName}" (${index})` });
-					return Promise.reject(error);
-				});
-
-				return await apiService.getDocumentByKey(bucketName, keyBody.key).catch(error => {
-					if (numberOfAttempts >= maxNumberOfAttempts) {
-						throw error;
-					}
-
-					logger.error({
-						message: `Error fetching document by random key from bucket "${bucketName}" (${index})`,
-					});
-
-					numberOfAttempts++;
-				});
-			},
-			(error, result) => {
-				if (error) {
-					logger.error(error);
-					return reject(error);
-				}
-
-				logger.info(`Successfully read ${result.length} document(s)`);
-
-				const documents = (result || [])
-					.filter(item => Boolean(item) && !isBinaryFile(item))
-					.map(item => ({ [bucketName]: item.json }));
-
-				return resolve(documents);
-			},
-		);
-	});
 };
 
+/**
+ * @param {{ connectionInfo: ConnectionInfo; logger: Logger; app: App }} param0
+ * @returns {Promise<object[]>}
+ */
 const getIndexes = async ({ connectionInfo, logger, app }) => {
 	logger.info(`Start getting indexes using REST API`);
 
 	const apiService = createRestApiService({ connectionInfo, app });
```

Comment:
Should we try-catch here and just return empty array if something went wrong?

Reply:
When getting indexes, we initially want to get them through restAPi, but if an error occurs here, then we make a second attempt to get indexes using n1ql
https://github.com/hackolade/CouchbaseV7Plus/blob/c641d44d6b576126ac300c4840ba1162a2e22cae/reverse_engineering/helpers/indexHelper.js#L112-L130